tt_metal/detail/tt_metal.hpp

```diff
@@ -553,7 +553,14 @@ namespace tt::tt_metal{
                         if (device_id != device->id()) {
                             // This means the issue queue and completion queue interfaces that service a remote device are being set up
                             // the issue queue interface needs to send fast dispatch packets to the "src" ethernet core
-                            consumer_physical_core = device->ethernet_core_from_logical_core(*device->get_active_ethernet_cores().begin());
+                            CoreCoord logical_eth_router_src =
+                                tt::Cluster::instance().get_and_configure_corresponding_eth_core_for_device(
+                                    device->id(), issue_q_physical_core, EthRouterMode::FD_SRC, device_id);
+                            consumer_physical_core = device->ethernet_core_from_logical_core(logical_eth_router_src);
+                            // TODO: remote processor core should use logical_eth_router_dst as producer
+                            CoreCoord logical_eth_router_dst =
+                                tt::Cluster::instance().get_and_configure_corresponding_eth_core_for_device(
+                                    device_id, issue_q_physical_core, EthRouterMode::FD_DST, device->id());
```

Comment:
you can remove this from here, if you look above this case is for programming the MMIO device with dispatch kernels. I can grab the eth dst router when setting up the program for the remote device

Reply:
yea I was thinking about this.. what happens if we have four ethernet links like we do in Galaxy, and between two chips we have two cores as SRC and two as DST. Wouldn't it be more correct to somehow set up the DST by just grabbing the connected ethernet core for the selected eth SRC?

Reply:
when we configure the remote dispatch program we can find which chip the remote issue queue reader is on and then get the eth router associated with that core which will give us src router. From the src we can get the corresponding dst? and ensure this matches to the eth router that feeds the remote processor